src/main/java/org/quiltmc/loader/impl/QuiltLoaderImpl.java

```diff
@@ -694,6 +694,16 @@ public void appendModTable(Consumer<String> to) {
 			}
 		}
 
+
+		HashMap<String, Set<String>> types = new HashMap<>();
+
+		for (ModContainerExt mod : mods) {
+			types.computeIfAbsent(mod.pluginId(), k -> new HashSet<>()).add(mod.modType());
+		}
+
+		table.addExtraData("Mod Table Version: 2");
+		table.addExtraData("Plugin Types: " + types);
```

Comment:
I don't really like the idea of "extra data" being added on to a table - especially since you can just emit those lines directly? (In the future we could turn the plugin types into a proper table too - which would be better off as its own table rather than appended to the existing table)